src/material/schematics/ng-generate/mdc-migration/rules/style-migrator.ts

```diff
@@ -102,9 +111,43 @@ export abstract class StyleMigrator {
   replaceLegacySelector(rule: postcss.Rule): void {
     for (let i = 0; i < this.classChanges.length; i++) {
       const change = this.classChanges[i];
-      if (rule.selector.includes(change.old)) {
+      if (rule.selector.match(change.old + END_OF_SELECTOR_REGEX)) {
         rule.selector = rule.selector.replace(change.old, change.new);
```

Comment:
(In this example, `mat-chip` would be the legacy selector)

Reply:
I added this case to one of the tests. I also just ended up changing the regular expression to just exclude matches where '-' follows the selector since that was simpler